apps/src/lab2/views/components/Instructions/ContinueButton.tsx

```diff
@@ -0,0 +1,73 @@
+import {
+  Button,
+  ButtonType,
+  ButtonColor,
+} from '@code-dot-org/component-library/button';
+import {ComponentSizeXSToL} from '@code-dot-org/component-library/common/types';
+import {FontAwesomeV6IconProps} from '@code-dot-org/component-library/fontAwesomeV6Icon';
+import React from 'react';
+
+import WithConditionalTooltip from '@cdo/apps/codebridge/components/WithConditionalTooltip';
+import continueOrFinishLesson from '@cdo/apps/lab2/progress/continueOrFinishLesson';
+import {useAppDispatch} from '@cdo/apps/util/reduxHooks';
+
+import moduleStyles from '@cdo/apps/lab2/views/components/Instructions/instructions.module.scss';
+
+interface ContinueButtonProps {
+  disabled: boolean;
+  type: ButtonType;
+  color: ButtonColor;
+  iconRight: FontAwesomeV6IconProps | undefined;
+  text: string;
+  tooltipMessage?: string;
+  hidden?: boolean;
+  size?: ComponentSizeXSToL;
+  className?: string;
+}
+
+/**
+ * Displays the 'Continue' or 'Finish' button that advances to the next level or finishes the progression unless
+ * hidden is true.
+ */
+const ContinueButton: React.FC<ContinueButtonProps> = ({
+  disabled,
+  type,
+  color,
+  iconRight,
+  text,
+  tooltipMessage,
+  hidden,
+  size,
+  className,
```

Comment:
optional: all the props here besides `tooltipMessage` and `hidden` are passed straight through to `<Button>`, so to simplify you could to `{tooltipMessage, hidden, ...buttonProps}` and just pass through `buttonProps`.

The `ContinueButtonProps` interface could also optionally just be `ButtonProps & {tooltipMessage?: string; hidden?: boolean}` - but this is fine too if we want to be more specific about which button props are allowed.